File: 283 Move Zeroes/283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if not nums or len(nums) == 1:
            return
        slow = 0
        fast = 0
        while fast < len(nums):
            if nums[fast] != 0:
                if slow != fast:
                    nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1

File: 283 Move Zeroes/283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    if (!nums || nums.length == 1) {
        return;
    }

    let slow = 0, fast = 0;
    while (fast < nums.length) {
        if (nums[fast] !== 0) {
            if (slow !== fast) {
                [nums[slow], nums[fast]] = [nums[fast], nums[slow]]
            }
            slow++
        }
        fast++
    }
};